packages/hash/frontend/src/blocks/page/createSuggester/useFilteredBlocks.tsx

```diff
@@ -0,0 +1,41 @@
+import { BlockVariant } from "blockprotocol";
+import { useMemo } from "react";
+
+import { UserBlock } from "../../userBlocks";
+import { fuzzySearchBy } from "./fuzzySearchBy";
+
+export const useFilteredBlocks = (
+  searchText: string,
+  userBlocks: UserBlock[],
+) => {
+  return useMemo(() => {
+    const allOptions: {
+      variant: BlockVariant;
+      meta: UserBlock;
+    }[] = Object.values(userBlocks).flatMap((blockMeta) =>
+      blockMeta.variants
+        ? blockMeta.variants.map((variant) => ({
+            variant: {
+              ...variant,
+              name: variant.name ?? variant.displayName,
+            },
+            meta: blockMeta,
+          }))
+        : {
+            variant: {
+              description: blockMeta.description,
+              name: blockMeta.displayName,
+              icon: blockMeta.icon,
+              properties: {},
+            } as BlockVariant,
```

Comment:
What is this cast necessary for?

Reply:
The TS compiler was printing a long, unreadable error here. This typecase helps in narrowing down the error search.

Reply:
Is it possible that error is real and your cast is hiding it? I feel like we need a stronger justification for a cast than this.

Reply:
Yeah, I misunderstood how casts should be used. I re-implemented the above piece of code using explicit type definitions [here](https://github.com/hashintel/hash/pull/407/files#diff-3c4af471fe6ad4e0f22f500c987f822b65b37f0a342a72e7f9de7650964cd21eR7).